core/domain/story_domain.py

```diff
@@ -272,8 +275,12 @@ def validate(self):
         for node_id in self.destination_node_ids:
             if not isinstance(node_id, basestring):
                 raise utils.ValidationError(
-                    'Expected each node id to be a string, ' +
+                    'Expected each node id to be a string, '
                     'received %s' % node_id)
+            if node_id == self.id:
+                raise utils.ValidationError(
+                    'Expected destination node id list of a node to not have '
```

Comment:
The wording here is a bit clumsy. How about: "The story node with ID {{node_id}} points to itself."?

Also, re:

> Generally, nodes are unlocked when the prerequisites are acquired, right? So, in the check for cyclic, should that be taken into account also?

Yes, I think so. One way you could do it is to simulate a user starting with "all prereq skills - all acquired skills" (presumably they've acquire the prereqs outside the story since there's nowhere else to do so). Then see if you can complete all the lessons using that as a basis, augmenting that list of skills along the way. I think we did something similar for collections, you might want to take a look at the code there too.

Reply:
Done, I added the validation and a test that tests three cases for this, can you check if that's fine?